### PR TITLE
feat (DOCS-CODE-010): [client-crendential-flow] .NET (C#) console app canonical sample

### PR DESCRIPTION
**Pull request by ferantivero**

### CODE REVIEW

**Review by skabou — COMMENTED** [review]
@ferantivero This is totally a nit but I see you have the name of the app as 'active-directory-dotnet-console-app-client-credentail-flow'

The 2nd to last word is misspelled.  Should be 'credential'

It also appears in the JSON output.  Just FYI

File: src/cli-access-protected-api/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  "displayName": "active-directory-dotnet-console-app-client-credentail-flow",
+  "description": null,
+  "groupMembershipClaims": null,
+  ...
+}
+```
+
+## About the code
+
+This .NET (C#) console  application uses a client secret as its credentials to retrieve an access token that's scoped for the Microsoft Graph API, and then uses that token to access its own application registration information.
```
Comment (skabou) [inline src/cli-access-protected-api/README.md:109]:
```suggestion
This .NET (C#) console application uses a client secret as its credentials to retrieve an access token that's scoped for the Microsoft Graph API, and then uses that token to access its own application registration information.
```

**Comment by ferantivero** [comment]
thanks for feedback @skabou all-in

**Review by skabou — APPROVED** [review]
Looks great! 🚀 🚢

**Review by ckittel — APPROVED** [review]
Only really small changes.  Nice!

File: src/cli-access-protected-api/Program.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    ClientId = "",
+    // Client secret 'Value' (not its ID) from 'Client secrets' in app registration in Azure portal
+    ClientSecret = "",
+    // Client 'Object ID' of app registration in Azure portal - this value is a GUID
+    ClientObjectId = ""
+};
+
+// This app instance should be a long-lived instance because
+// it maintains the in-memory token cache.
+IConfidentialClientApplication app = ConfidentialClientApplicationBuilder.Create(config.ClientId)
```
[the remark this answers is not in the document]
Comment (ferantivero) [reply src/cli-access-protected-api/Program.cs]: sure thing @ckittel 

done |  address from [d7aed](https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/28/commits/d7aed6a46aa75d36d3c9415f620899cc40e47216)

File: src/cli-access-protected-api/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  ...
+}
+```
+
+## Prerequisites
+
+- An Azure Active Directory (Azure AD) tenant. You can [open an Azure account for free](https://azure.microsoft.com/free) to get an Azure AD instance.
+- [.NET 6.0 SDK](https://dotnet.microsoft.com/download/dotnet/6.0)
+
+## Setup
```
[the remark this answers is not in the document]
Comment (ferantivero) [reply src/cli-access-protected-api/README.md:51]: a shot in the dark below @ckittel, not sure if I followed.

a fix | attempt to address [ef684](https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/28/commits/ef684d2ed06f3d8c247c5c64a76fb1b0b09e0747)

[the remark this answers is not in the document]
Comment (ckittel) [reply src/cli-access-protected-api/README.md:51]: I'm blind.  Somehow I completely missed it above.  No changes necessary (your change you made was fine as well, no need to revert).  Sorry about that.  Friday morning brain fart :)

**Comment by ferantivero** [comment]
@mmacy a new review from content dev is requested over here.

**Review by mmacy — APPROVED** [review]
Couple nits, but LGTM. :shipit

File: src/cli-access-protected-api/Program.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.Identity.Client;
+using Microsoft.Identity.Web;
+using System.Net.Http.Headers;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+
+var config = new {
```
Comment (mmacy) [inline src/cli-access-protected-api/Program.cs:7]:
```suggestion
var _config = new {
```

**Comment by ferantivero** [comment]
@ckittel please let's go ahead a merge based on your convenience

**Pull request merged**